src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
```

Comment:
No, it's the same logic - the only difference is that the first call is to Assembly.Load(String) rather than Load(AssemblyName). You could use the common function just by creating an AssemblyName out of the string "assemblyName" first (that's just what Assembly.Load(string) does internally so you're not adding overhead by doing this.)